.github/workflows/publish-artifacts.yml

```
@@ -0,0 +1,174 @@
+name: Publish artifacts to ORAS
+
+on:
+  push:
+    branches:
+    - main
+
+env:
+  RUST_TOOLCHAIN: 1.80.0
+
+jobs:
+  publish-aa:
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
+    strategy:
+      matrix:
+        tee:
+        - none
+        - amd
+        - az-cvm-vtpm
+        - tdx
+        - se
+        - cca
+        include:
+        - tee: none
+          libc: musl
+          arch: x86_64
+        - tee: amd
+          libc: musl
+          arch: x86_64
+        - tee: az-cvm-vtpm
+          libc: gnu
+          arch: x86_64
+        - tee: tdx
+          libc: gnu
+          arch: x86_64
+        - tee: se 
+          libc: gnu
+          arch: s390x
+        - tee: cca
+          libc: musl
+          arch: x86_64
+    runs-on: ${{ matrix.arch == 's390x' && 's390x' || 'ubuntu-22.04' }}
+    env:
+      TEE_PLATFORM: ${{ matrix.tee }}
+      LIBC: ${{ matrix.libc }}
+      REGISTRY: ghcr.io
+      IMAGE_NAME: ${{ github.repository }}
+    steps:
+    - name: Take a pre-action for self-hosted runner
+      if: matrix.arch == 's390x'
+      run: |
+        if [ -f "${HOME}/script/pre_action.sh" ]; then
+          "${HOME}/script/pre_action.sh" cc-guest-components
+        fi
+
+    - name: Log in to the Container registry
+      uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - uses: oras-project/setup-oras@v1
+      with:
+        version: 1.0.0
+
+    - uses: actions/checkout@v4
+
+    - uses: actions-rust-lang/setup-rust-toolchain@v1
+      with:
+        toolchain: ${{ env.RUST_TOOLCHAIN }}
+        target: ${{ matrix.arch }}-unknown-linux-${{ matrix.libc }}
+        override: true
+        components: rustfmt, clippy
+
+    - name: Install tpm dependencies
+      if: matrix.tee == 'az-cvm-vtpm'
+      run: |
+        sudo apt-get install -y --no-install-recommends libtss2-dev
+
+    - name: Install tdx dependencies
+      if: matrix.tee == 'tdx'
+      run: |
+        sudo curl -sL https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | sudo gpg --dearmor --output /usr/share/keyrings/intel-sgx.gpg
+        sudo echo 'deb [arch=amd64 signed-by=/usr/share/keyrings/intel-sgx.gpg] https://download.01.org/intel-sgx/sgx_repo/ubuntu jammy main' | sudo tee /etc/apt/sources.list.d/intel-sgx.list
+        sudo apt-get update
+        sudo apt-get install -y --no-install-recommends libtdx-attest-dev
```

we've started to have the vTPM/DCAP installation copied in many places so perhaps we could make them reusable in workflows using 
```suggestion
      uses: "./.github/workflows/lib-install-dcap.yaml"
```

---

oof, I don't know. I'd personally prefer some duplication instead of excessive abstraction and people having to jump through workflows to replicate what's happening. for installing tpm headers this would be definitely overkill but having a reusable wf for the tdx dep might be tolerable. I suppose such a workflow needs to be parameterized per ubuntu distro?

---

ah, wait. that doesn't make sense. a workflow will run in its own worker machine, so you'd be installing packages in the child wf , but it wouldn't be available in the calling workflow

---

but we can use a "composite" action for sharing steps, added a commit.

---

I have no strong opinions. My comment was based on the fact that the DCAP installation is a bit broken and I was hoping to do the fixing in one place only. OTOH, `libtdx-attest-dev` is eventually going away so maybe I'll just wait for that to happen instead..